src/ipfsd-daemon.js

```diff
@@ -57,6 +57,14 @@ class Daemon {
     this._gatewayAddr = null
     this._started = false
     this.api = null
+    this.keySize = null
+
+    this.keySize = process.env.IPFS_KEYSIZE
+
+    // option takes precedence over env variable
```

Comment:
I don't think this is right. First it should use the default value. Second it should use the passed in value and lastly, if there is a environment variable, use that. 

This way we can override application configs when deploying etc.

Reply:
👍 for what @VictorBjelkholm said.

Reply:
Wait, what @VictorBjelkholm said reads as - default take precedence over env, which takes precedence over option? 

Seems backwards, unless I'm reading it wrong. In any case, this is what I'm doing now - option takes precedence over ENV variable, which takes precedence over default. I think that is the canonical way of doing it and allows passing defaults with the ENV variable, but override them on a case by case basis in tests cases and elsewhere?

Reply:
`Default -> Options trumps Default -> Env trumps Options and Default`

Reply:
@dryajov I believe you are using `precedence` as "it comes before === less value" but the english definiton says "precedence: the condition of being considered more important than someone or something else; priority in importance, order, or rank." and so, if default took precedence over env, default values would always win.

Reply:
> option takes precedence over ENV variable, which takes precedence over default.

Which uses `precedence` in the english definition of the term.

I think the contention is whether ENV take precedence over option or vice versa. The case I'm making is that an ENV variable will define a global that should be overridable through an option when calling `spawn`. This allows to override the default globally and locally on a case by case basis.

Reply:
So in my case it is -

`Default -> Env trumps Default -> Options trumps Default and Env`

But I'm not dead set on it, but it feels more natural to me.

Reply:
@dryajov use https://github.com/ipfs/js-ipfsd-ctl/pull/203#discussion_r168822236, which is pretty much how every system works. 

I wonder which system are you familiar with that uses it the other way around.

Reply:
Any programming language with correct scoping rules :)